### PR TITLE
DSPDC-163 Add result listener.

### PR DESCRIPTION
**Pull request by danxmoran**
Still TODO (in a follow-up PR): Construct the listener & run it as part of app startup.

### CODE REVIEW

File: manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala [hunk 1]
```diff
@@ -223,29 +246,54 @@ object DbClient {
       sql"""delete from transfer_requests where id = $id""".update.run.void
         .transact(transactor)
 
+    override def updateTransfers(summary: List[(UUID, TransferSummary)]): IO[Unit] = {
+      val newStatuses = summary.map {
+        case (id, s) =>
+          val status = s.result match {
+            case TransferResult.Success          => TransferStatus.Succeeded
```
Comment (jacarey) [inline manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala]: Didn't transfer status go away?

File: manager/src/main/scala/org/broadinstitute/transporter/transfer/ResultListener.scala [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      val logPrefix = s"Recording ${results.length} transfer results"
+
+      for {
+        _ <- if (numMalformed > 0) {
+          logger.warn(s"$logPrefix; ignoring $numMalformed malformed entities in batch")
+        } else {
+          logger.info(logPrefix)
+        }
+        _ <- dbClient.updateTransfers(results)
+        _ <- resubmitTransientFailures(results)
```
Comment (jacarey) [inline manager/src/main/scala/org/broadinstitute/transporter/transfer/ResultListener.scala]: How do we limited the number of times we will attempted to retry a transient failure? Is there a kill switch?

File: manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala [hunk 3]
```diff
@@ -223,29 +246,54 @@ object DbClient {
...
-              values($requestId, $queueId, ${transferIds.length})""".update.run.void
+        sql"""insert into transfer_requests (id, queue_id) values ($requestId, $queueId)""".update.run.void
 
-      val insertTransfers = Update[UUID](
-        s"insert into transfers (id, request_id, status) values (?, '$requestId', 'submitted')"
-      ).updateMany(transferIds).void
+      val insertTransfers = Update[(UUID, Json)](
+        s"""insert into transfers (id, request_id, status, body, info)
+            values (?, '$requestId', '${TransferStatus.Submitted.entryName}', ?, NULL)"""
+      ).updateMany(transfersById).void
```
Comment (jacarey) [inline manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala]: Should we sanity check the insert return to make sure it says it inserted the same number we submitted?

File: manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala (hunk 1 above)
Comment (jacarey) [reply manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala]: nevermind i see it now

File: manager/src/main/scala/org/broadinstitute/transporter/transfer/ResultListener.scala (hunk 2 above)
Comment (danxmoran) [reply manager/src/main/scala/org/broadinstitute/transporter/transfer/ResultListener.scala]: There isn't one yet. I could add something hard-coded for now just so the machinery is there.

**Pull request merged**